crates/astria-gossipnet/src/network.rs

```diff
@@ -95,10 +95,24 @@ impl NetworkBuilder {
         self
     }
 
+    /// Specify the keypair to use from a file.
+    /// The file should contain a hex-encoded 32-byte ed25519 secret key.
+    pub fn keypair_from_file<P: AsRef<std::path::Path>>(mut self, path: P) -> Result<Self> {
+        let key_string = std::fs::read_to_string(path).wrap_err("failed to read keypair file")?;
+        if key_string.len() < 64 {
```

Comment:
Is the extra check here necessary? Wouldn't this fail if just calling `hex::decode(&key_string)` and then `Keepair::ed25519_from_bytes(bytes)`?

If you decide to keep it, prefer `color_eyre::eyre::ensure!`:
```suggestion
        ensure!(
            key_string.len() >= 64,
            "keypair file is too short"
        );
```

Reply:
i wanted to pass exactly 64 bytes into `hex::decode` because sometimes people have a newline/space at the end of the file which causes hex::decode to fail if that's written in